interfaces/acados_template/acados_template/casadi_function_generation.py

```diff
@@ -571,6 +560,7 @@ def generate_c_code_conl_cost(model: AcadosModel, stage_type: str, opts):
     fun_name_cost_fun_jac_hess = model.name + suffix_name_fun_jac_hess
 
     # set up functions to be exported
+    # TODO: fix these for p_global
```

Comment:
what is missing here? maybe add to PR description?

Reply:
I believe that this commit https://github.com/acados/acados/pull/1234/commits/3e2be45062243d44c6baeef8c82957591fc7adec should fix the code generation also if the outer loss depends on `p_global`.
Before that commit I would expect that CasADi complains about free variables.